ragna/core/_rag.py

```diff
@@ -232,9 +235,14 @@ async def answer(self, prompt: str, *, stream: bool = False) -> Message:
 
         return answer
 
-    def _parse_documents(self, documents: Iterable[Any]) -> list[Document]:
-        documents_ = []
-        for document in documents:
+    def _parse_input(
```

Comment:
@nenb @blakerosenthal Loosely leaning on https://github.com/Quansight/ragna/issues/256#issuecomment-1954057313, `input` is overloaded with 3 values:

1. If its explicitly `None`, i.e. no default value and actually calling `Chat(input=None)`, we don't want any metadata filtering and thus also `metadata_filter=None`. When we pass this down to the source storage, filtering should be disabled and thus the whole index should be used. _This is not implemented yet in this PR_
2. If its a `MetadataFilter`, we use it.
3. If its anything else, we assume it is list of documents or paths to create documents from. For a better UX, we might also allow a single paths or document, but this is not critical.

For 1. and 2. we assume the chat is already prepared and only require preparation for 3.

Reply:
@pmeier @nenb What remains on the source storage work in particular? It looks like the chat object performs the logic of constructing a `MetadataFilter` object to pass along to source storage, and the source storage needs to handle the `metadata_filter=None` case and return the entire index. Is there anything else?

Reply:
Not sure if that is implied, but we need handling of the `MetadataFilter` in `Chroma` and `LanceDb`. But I think these are the open tasks right now.

Reply:
@pmeier @blakerosenthal I had some uncertainty in the task as well.

I've opened #460. We can discuss what needs to be done there. And @blakerosenthal you can push the `lancedb` changes to the same branch.